# **Exploring data from my Fantasy Football Manager Rating project**

**Let's start by examing the lowest rating:**

In [3]:
SELECT team, ffmr, week_of_season, season 
FROM elo.timeline 
ORDER BY ffmr 
LIMIT 1

(1 row(s) affected)

Total execution time: 00:00:01.006

team ffmr week_of_season season Sean 836.9054042 6 2022

Taking a closer look at the beginning of the 2022 season, Sean entered the year with a 868.95 rating and even defeated a near 1100 rated opponent in week 1. Unfortunately for him, he scored below the median in week 1, so his rating didn't increase 

by that much. Following that, he lost his next 4 matchups, plummeting his rating to the lowest in league history.

In [41]:
SELECT team, ffmr, score, opponent, opponent_ffmr, winner, week_of_season, season
FROM elo.timeline
WHERE team = "Sean" and season = '2022'
LIMIT 7

(7 row(s) affected)

Total execution time: 00:00:01.030

team ffmr score opponent opponent_ffmr winner week_of_season season Sean 868.9462139 113.78 John 1099.323164 Sean 1 2022 Sean 875.0141916 64.44 Loof 1129.54885 Loof 2 2022 Sean 866.2581202 109.5 Nick 1095.120116 Nick 3 2022 Sean 857.1021804 86.62 Nate 1051.860391 Nate 4 2022 Sean 848.2435304 107.04 Shev 923.9621109 Shev 5 2022 Sean 836.9054042 120.8 Justin 936.805472 Sean 6 2022 Sean 858.2465871 119.56 Pfeff 929.1462149 Sean 7 2022

Here's a bar graph of Sean's rating, so you can see the highs and lows. It appears that he'll be entering into next season with a rating of around 900, let's hope he can make a return to his previous form and start competing for the championship again.

(You can find more detailed visualizations on my Tableau Public profile: https://public.tableau.com/app/profile/john.bolger )

In [39]:
SELECT ffmr
FROM elo.timeline
WHERE team = "Sean"

(76 row(s) affected)

Total execution time: 00:00:01.006

ffmr 1016.0 1017.001196 1032.860132 1048.271492 1030.563174 1044.177398 1056.997747 1070.830017 1083.130526 1095.747795 1109.574896 1121.409774 1087.892711 1068.929856 1072.43691 1051.95435 1033.017055 1048.652979 1044.154361 1026.072846 1022.636727 1038.735512 1020.753779 1035.045174 1016.470377 1031.806778 1045.781546 1027.6267 1010.52747 987.6894973 972.6050383 956.4017015 958.0696085 977.2477997 962.8812738 948.437185 933.7057925 920.4967705 909.5693409 897.0465147 884.4440656 888.6801555 879.2400453 867.1186603 870.0634374 892.364888 880.9312381 871.3117786 893.0221418 916.1102815 904.5961527 892.6420615 897.5143327 898.0384918 887.7455088 908.3223305 897.4882383 887.0388245 877.3773574 865.8804839 868.9462139 875.0141916 866.2581202 857.1021804 848.2435304 836.9054042 858.2465871 879.0791764 886.0698818 875.3899783 898.6773628 889.2201579 895.2892317 885.2927007 872.2541967 892.5651384

Sean actually holds the 24 lowest ratings in league history... yikes

In [55]:
SELECT team, ffmr, week_of_season, season 
FROM elo.timeline 
ORDER BY ffmr 
LIMIT 25

(25 row(s) affected)

Total execution time: 00:00:01.006

team ffmr week_of_season season Sean 836.9054042 6 2022 Sean 848.2435304 5 2022 Sean 857.1021804 4 2022 Sean 858.2465871 7 2022 Sean 865.8804839 16 2021 Sean 866.2581202 3 2022 Sean 867.1186603 15 2020 Sean 868.9462139 1 2022 Sean 870.0634374 1 2021 Sean 871.3117786 4 2021 Sean 872.2541967 15 2022 Sean 875.0141916 2 2022 Sean 875.3899783 10 2022 Sean 877.3773574 15 2021 Sean 879.0791764 8 2022 Sean 879.2400453 14 2020 Sean 880.9312381 3 2021 Sean 884.4440656 12 2020 Sean 885.2927007 14 2022 Sean 886.0698818 9 2022 Sean 887.0388245 14 2021 Sean 887.7455088 11 2021 Sean 888.6801555 13 2020 Sean 889.2201579 12 2022 Shev 890.6621515 14 2022

**Now let's take a look at the highest rating:**

In [4]:
SELECT team, ffmr, week_of_season, season 
FROM elo.timeline 
ORDER BY ffmr DESC 
LIMIT 1

(1 row(s) affected)

Total execution time: 00:00:01.004

team ffmr week_of_season season Loof 1138.030319 17 2021

Coming of a 2020 championship, Loof stayed hot and hit his peak at the very end of the 2021 season gaining over 50 rating points in the final 5 weeks to reach a league record 1138 rating. Unfortunatly, Loof failed to repeat as he ended up losing the

the championship matchup to John.

In [44]:
SELECT team, ffmr, score, opponent, opponent_ffmr, winner, week_of_season, season
FROM elo.timeline
WHERE team = "Loof" and season = '2021'

(17 row(s) affected)

Total execution time: 00:00:01.005

team ffmr score opponent opponent_ffmr winner week_of_season season Loof 1126.672835 96.5 Shev 973.6575368 Shev 1 2021 Loof 1104.510694 99.56 Nate 948.7199719 Nate 2 2021 Loof 1083.186308 143.62 Sean 880.9312381 Loof 3 2021 Loof 1093.826834 102.02 John 1108.05934 John 4 2021 Loof 1076.065872 149.5 Justin 956.6146724 Loof 5 2021 Loof 1086.92981 137.62 Eddie 1034.748908 Loof 6 2021 Loof 1099.078835 126.36 Derek 1010.526906 Derek 7 2021 Loof 1097.031411 143.96 Nick 1115.326439 Loof 8 2021 Loof 1110.738442 99.66 Pfeff 978.3905395 Pfeff 9 2021 Loof 1088.928661 123.94 Shev 967.9345236 Loof 10 2021 Loof 1100.805947 121.06 Nate 996.2688854 Nate 11 2021 Loof 1081.408355 135.0 Sean 908.3223305 Loof 12 2021 Loof 1092.428266 123.8 John 1085.055321 Loof 13 2021 Loof 1106.14505 134.92 Justin 954.6775339 Loof 14 2021 Loof 1115.845959 99.2 Pfeff 969.6106143 Loof 15 2021 Loof 1126.096667 122.46 Nate 1018.427868 Loof 16 2021 Loof 1138.030319 130.8 John 1098.298588 John 17 2021

Loof's rating: (he entered the league in 2020 so his graph is a bit smaller than Sean's)

In [30]:
SELECT ffmr
FROM elo.timeline
WHERE team = "Loof"

(48 row(s) affected)

Total execution time: 00:00:01.006

ffmr 1000.0 1014.817261 1031.564491 1046.432341 1045.459748 1059.731065 1073.623999 1055.298764 1068.096245 1081.213006 1092.63323 1091.105741 1102.517452 1120.063231 1131.812677 1126.672835 1104.510694 1083.186308 1093.826834 1076.065872 1086.92981 1099.078835 1097.031411 1110.738442 1088.928661 1100.805947 1081.408355 1092.428266 1106.14505 1115.845959 1126.096667 1138.030319 1119.62927 1129.54885 1120.997976 1100.306926 1081.845423 1060.802083 1075.915645 1089.049008 1084.182904 1095.431819 1088.613889 1081.74366 1063.17003 1063.855908 1075.393828 1057.22446

Some other top ratings:

In [51]:
SELECT team, ffmr, week_of_season, season 
FROM elo.timeline 
ORDER BY ffmr DESC 
LIMIT 10

(10 row(s) affected)

Total execution time: 00:00:01.013

team ffmr week_of_season season Loof 1138.030319 17 2021 Loof 1131.812677 16 2020 Loof 1129.54885 2 2022 Loof 1126.672835 1 2021 Loof 1126.096667 16 2021 John 1123.640596 16 2022 Sean 1121.409774 13 2018 Loof 1120.997976 3 2022 Loof 1120.063231 15 2020 Loof 1119.62927 1 2022

**Notable Matchups (Matchups are displayed twice, sorry it doesn't look very nice)**

Here are the top 5 matchups with the highest combined ratings. The aforementioned 2021 championship game tops the list, which shouldn't be too surprising. Note that Loof is in every single one of them and that 3 of the 5 are matchups between 

Loof and John. It's also interesting to note that John beat Loof all 3 times.

In [52]:
SELECT team, ffmr, opponent, opponent_ffmr, week_of_season, season, winner
FROM elo.timeline 
ORDER BY ffmr + opponent_ffmr DESC 
LIMIT 10

(10 row(s) affected)

Total execution time: 00:00:01.012

team ffmr opponent opponent_ffmr week_of_season season winner John 1098.298588 Loof 1138.030319 17 2021 John Loof 1138.030319 John 1098.298588 17 2021 John Loof 1097.031411 Nick 1115.326439 8 2021 Loof Nick 1115.326439 Loof 1097.031411 8 2021 Loof Eddie 1087.227376 Loof 1120.063231 15 2020 Loof Loof 1120.063231 Eddie 1087.227376 15 2020 Loof Loof 1093.826834 John 1108.05934 4 2021 John John 1108.05934 Loof 1093.826834 4 2021 John Loof 1100.306926 John 1100.527304 4 2022 John John 1100.527304 Loof 1100.306926 4 2022 John

Now let's take a look at the biggest stinkers, all featuring Sean (of course). I won't even show the winners becuase who really cares.

In [57]:
SELECT team, ffmr, opponent, opponent_ffmr, week_of_season, season
FROM elo.timeline 
ORDER BY ffmr + opponent_ffmr
LIMIT 10

(10 row(s) affected)

Total execution time: 00:00:01.017

team ffmr opponent opponent_ffmr week_of_season season Shev 923.9621109 Sean 848.2435304 5 2022 Sean 848.2435304 Shev 923.9621109 5 2022 Sean 836.9054042 Justin 936.805472 6 2022 Justin 936.805472 Sean 836.9054042 6 2022 Sean 885.2927007 Shev 890.6621515 14 2022 Shev 890.6621515 Sean 885.2927007 14 2022 Pfeff 929.1462149 Sean 858.2465871 7 2022 Sean 858.2465871 Pfeff 929.1462149 7 2022 Nate 922.764491 Sean 867.1186603 15 2020 Sean 867.1186603 Nate 922.764491 15 2020

Heres the matchups with the greatest rating disparities, all vs Sean (of course).

In [11]:
SELECT team, ffmr, opponent, opponent_ffmr, week_of_season, season, winner
FROM elo.timeline 
ORDER BY ffmr - opponent_ffmr DESC
LIMIT 10

(10 row(s) affected)

Total execution time: 00:00:01.015

team ffmr opponent opponent_ffmr week_of_season season winner Loof 1129.54885 Sean 875.0141916 2 2022 Loof John 1099.323164 Sean 868.9462139 1 2022 Sean Nick 1095.120116 Sean 866.2581202 3 2022 Nick John 1088.939854 Sean 875.3899783 10 2022 Sean John 1103.230695 Sean 893.0221418 5 2021 Sean Loof 1083.186308 Sean 880.9312381 3 2021 Loof John 1086.615893 Sean 888.6801555 13 2020 John Nate 1051.860391 Sean 857.1021804 4 2022 Nate Loof 1088.613889 Sean 898.6773628 11 2022 Loof Nick 1057.626097 Sean 870.0634374 1 2021 Sean

**Ratings by team**

Let's take a look at the ratings for each manager. Note that the top 4 teams in terms of average rating are the 4 teams with championships and that Sean's actually not last for once.

In [62]:
SELECT team, AVG(ffmr), MIN(ffmr), MAX(ffmr)
FROM elo.timeline
GROUP BY team 
ORDER BY AVG(ffmr) DESC

(13 row(s) affected)

Total execution time: 00:00:01.010

team AVG(ffmr) MIN(ffmr) MAX(ffmr) Loof 1086.0377332291664 1000.0 1138.030319 John 1079.0640460657894 1016.0 1123.640596 Eddie 1032.1394608050002 995.6201383 1087.227376 Nick 1022.7993069683544 946.6285413 1115.326439 Derek 999.8766791513159 927.4672248 1103.420452 Nate 999.0650543118419 922.764491 1105.636317 Justin 988.9432105467531 907.0313124 1081.004377 Pete 983.7882646896553 947.0093265 1022.85791 Pfeff 978.2573317961038 897.5775829 1027.866558 Zach 974.231015775 939.0184708 1016.0 Sean 954.9696622026311 836.9054042 1121.409774 James 953.4395054800001 907.3629625 1001.4016 Shev 953.1638438246757 890.6621515 991.1617073

Now let's look at the most volitile seasons, the direction of the rating in these is not explicit, but can be inferred in some cases.

In [26]:
SELECT team, MIN(ffmr), MAX(ffmr), MAX(ffmr) - MIN(ffmr), season
FROM elo.timeline
GROUP BY team, season
ORDER BY MAX(ffmr) - MIN(ffmr) DESC
LIMIT 10

(10 row(s) affected)

Total execution time: 00:00:01.004

team MIN(ffmr) MAX(ffmr) MAX(ffmr) - MIN(ffmr) season Derek 966.8393456 1103.420452 136.58110640000007 2022 Loof 1000.0 1131.812677 131.8126769999999 2020 Sean 867.1186603 987.6894973 120.57083699999998 2020 Nick 999.1441087 1105.137995 105.9938863000001 2022 Sean 1016.0 1121.409774 105.40977399999997 2018 Nate 954.8916597 1058.60637 103.71471029999998 2018 Nate 922.764491 1020.253362 97.48887100000002 2020 James 907.3629625 1001.4016 94.03863750000005 2019 Justin 907.0313124 999.4709882 92.43967579999992 2021 Justin 991.8485601 1081.004377 89.15581689999999 2019

I think Derek's 2022 season is worthy of a deeper inspection. Coming off a rough year in 2021, Derek started the year with a 966.84 rating and absolutely killed it. His only regular season loses came against 1070+ rated opponents and he consistantly 

put up high scores. He reached his peak rating at the championship game and ended up losing, which seems to be a common occurance. A great season nonetheless.

In [63]:
SELECT team, ffmr, score, opponent, opponent_ffmr, winner, week_of_season, season
FROM elo.timeline
WHERE team = "Derek" and season = '2022'

(16 row(s) affected)

Total execution time: 00:00:01.011

team ffmr score opponent opponent_ffmr winner week_of_season season Derek 966.8393456 166.38 Nick 1086.126346 Derek 1 2022 Derek 985.6368651 146.58 John 1077.016372 John 2 2022 Derek 989.2765031 138.7 Justin 944.8651938 Derek 3 2022 Derek 1004.196916 152.02 Pfeff 936.1943462 Derek 4 2022 Derek 1018.812559 153.68 Eddie 1017.381627 Derek 5 2022 Derek 1034.874119 117.36 Nate 1079.037647 Nate 6 2022 Derek 1036.360742 103.54 Loof 1075.915645 Loof 7 2022 Derek 1020.102492 151.52 Sean 879.0791764 Derek 8 2022 Derek 1033.365593 134.9 Shev 932.057002 Derek 9 2022 Derek 1046.806708 120.3 Nick 1020.798901 Derek 10 2022 Derek 1061.12954 87.48 John 1084.447879 John 11 2022 Derek 1044.039491 148.26 Justin 987.0801425 Derek 12 2022 Derek 1056.868747 126.12 Pfeff 934.3839613 Derek 13 2022 Derek 1068.799009 166.38 Eddie 1043.339021 Derek 14 2022 Derek 1089.834797 143.08 Eddie 1025.980693 Derek 16 2022 Derek 1103.420452 75.42 Nick 1034.582526 Nick 17 2022

Here's the less interesting, least volitile seasons

In [27]:
SELECT team, MIN(ffmr), MAX(ffmr), MAX(ffmr) - MIN(ffmr), season
FROM elo.timeline
GROUP BY team, season
ORDER BY MAX(ffmr) - MIN(ffmr)
LIMIT 10

(10 row(s) affected)

Total execution time: 00:00:01.004

team MIN(ffmr) MAX(ffmr) MAX(ffmr) - MIN(ffmr) season Pfeff 992.1080077 1026.135904 34.02789629999995 2019 Nick 946.6285413 984.8730332 38.24449189999996 2019 Eddie 1020.937091 1063.674546 42.737454999999954 2019 Eddie 999.4774232 1043.339021 43.86159780000003 2022 John 1066.536098 1112.853612 46.317514000000074 2021 John 1077.016372 1123.640596 46.62422399999991 2022 Shev 942.9003157 989.9356585 47.03534280000008 2020 Pfeff 949.1485437 996.9431295 47.79458580000005 2021 Derek 978.1331 1026.091202 47.95810200000005 2019 Pfeff 979.5411632 1027.866558 48.32539479999991 2018

**Strength of Schedule**

In [75]:
SELECT team, AVG(opponent_ffmr)
FROM elo.timeline
GROUP BY team 
ORDER BY AVG(opponent_ffmr) DESC

(13 row(s) affected)

Total execution time: 00:00:01.008

team AVG(opponent_ffmr) Eddie 1011.787121 Nate 1008.3220809026313 Derek 1007.7721976500005 Zach 1007.45525259375 Sean 1006.3285160500001 Loof 1005.0905494083332 Pfeff 1004.9645384233768 James 1004.9053432200001 Nick 1003.8885925582277 Pete 1002.9222725068967 Shev 998.2378342090906 Justin 997.5591832623379 John 996.8494207513154

Hardest seasons:

In [80]:
SELECT team, AVG(opponent_ffmr), season
FROM elo.timeline
GROUP BY team, season 
ORDER BY AVG(opponent_ffmr) DESC
LIMIT 5

(5 row(s) affected)

Total execution time: 00:00:01.011

team AVG(opponent_ffmr) season Nate 1025.33055928125 2021 Eddie 1025.1246410294118 2022 Derek 1024.9846916333333 2020 Eddie 1023.1421816687499 2020 Nate 1023.0055554600001 2019

Easiest seasons: (Justin is the commishiner, seems a bit fishy to me)

In [81]:
SELECT team, AVG(opponent_ffmr), season
FROM elo.timeline
GROUP BY team, season 
ORDER BY AVG(opponent_ffmr)
LIMIT 5

(5 row(s) affected)

Total execution time: 00:00:01.011

team AVG(opponent_ffmr) season Justin 986.8627903624999 2022 Justin 987.38161545 2018 John 988.1944476466667 2019 John 990.4062408187499 2022 Nate 991.25694905 2018

Number of matchups vs 1100+ rated opponents: (Note that Derek hasn't had one even tho he's been in the league for a full 5 years)

In [77]:
SELECT team, COUNT(team)
FROM elo.timeline
WHERE opponent_ffmr > 1100
GROUP BY team

(9 row(s) affected)

Total execution time: 00:00:01.002

team COUNT(team) Sean 3 Eddie 6 Nate 9 Pfeff 3 Nick 4 Shev 3 Loof 3 Justin 4 John 2